### PR TITLE
chore(test): update testing libraries and refactor

### PR DESCRIPTION
**Pull request by ryo-manba**
## 📝 Description

Updated test-related libraries and refactor code.

Some warnings still appear during test execution, but I plan to address them separately.
Details of the changes are provided in the inline comments.

## 💣 Is this a breaking change (Yes/No):

No.

## Release Notes

- **New Features**
	- Improved user interaction testing across various components by utilizing the `userEvent` library for asynchronous event handling.
  
- **Bug Fixes**
	- Updated assertion methods in tests to align with Jest's recommended practices for clarity and consistency.
  
- **Documentation**
	- Enhanced test suite organization and readability by removing unnecessary `act` wrappers around user interactions.
  
- **Chores**
	- Updated import statements for testing libraries to reflect a more general React testing approach. 

These changes improve the overall reliability and maintainability of the application while ensuring a better testing experience.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ module.exports = {
     ],
   },
   transformIgnorePatterns: ["[/\\\\]node_modules[/\\\\].+\\.(js|jsx)$"],
-  setupFilesAfterEnv: ["@testing-library/jest-dom/extend-expect", "./scripts/setup-test.ts"],
+  setupFilesAfterEnv: ["@testing-library/jest-dom", "./scripts/setup-test.ts"],
```
Comment (ryo-manba) [inline jest.config.js:21]: `extend-expect` has been removed.
https://stackoverflow.com/questions/77328459/cannot-find-module-testing-library-jest-dom-extend-expect-from-jest-setup-js

File: packages/components/accordion/__tests__/accordion.test.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ describe("Accordion", () => {
 
     expect(() => wrapper.unmount()).not.toThrow();
 
-    expect(spy).toBeCalledTimes(0);
+    expect(spy).toHaveBeenCalledTimes(0);
```
Comment (ryo-manba) [inline packages/components/accordion/__tests__/accordion.test.tsx:33]: `toBeCalledTimes` is deprecated.
see: https://github.com/jestjs/jest/issues/13164

File: packages/components/accordion/__tests__/accordion.test.tsx [hunk 3]
```diff
@@ -121,9 +121,7 @@ describe("Accordion", () => {
 
     expect(button).toHaveAttribute("aria-expanded", "false");
 
-    await act(async () => {
-      await user.click(button);
-    });
+    await user.click(button);
```
Comment (ryo-manba) [inline packages/components/accordion/__tests__/accordion.test.tsx:124]: The warning no longer appears without wrapping `await user.click` in `act`.
This issue was caused by multiple versions of `@testing-library/dom` being present.
see: https://github.com/testing-library/user-event/discussions/906#discussioncomment-6165530

File: packages/components/select/__tests__/select.test.tsx [hunk 4]
```diff
@@ -858,18 +810,29 @@ describe("Select with React Hook Form", () => {
 
     wrapper = render(
       <form className="flex flex-col gap-4" onSubmit={handleSubmit(onSubmit)}>
-        <Select data-testid="select-1" items={itemsData} {...register("withDefaultValue")}>
+        <Select
+          data-testid="select-1"
+          items={itemsData}
+          {...register("withDefaultValue")}
+          aria-label="select-1"
```
Comment (ryo-manba) [inline packages/components/select/__tests__/select.test.tsx:817]: A warning from react-aria was shown when `aria-label` was missing.

File: plop/hook/__tests__/{{hookName}}.test.tsx.hbs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import {renderHook} from "@testing-library/react-hooks";
+import {renderHook} from "@testing-library/react";
```
Comment (ryo-manba) [inline plop/hook/__tests__/{{hookName}}.test.tsx.hbs:1]: Importing from `@testing-library/react-hooks` was causing a warning.
see: https://github.com/testing-library/react-hooks-testing-library/issues/832#issuecomment-1114321293

**Review by jrgarciadev — APPROVED** [review]

**Pull request merged**